src/shady/emit/c/emit_c.c

```diff
@@ -116,7 +116,7 @@ static bool has_forward_declarations(CDialect dialect) {
     }
 }
 
-static void emit_global_variable_definition(Emitter* emitter, AddressSpace as, String decl_center, const Type* type, bool constant, String init) {
+static void emit_global_variable_definition(Emitter* emitter, AddressSpace as, String decl_center, const Type* type, bool constant, String init, const Nodes* annotations) {
```

Comment:
Instead of passing annotations, add a "qualifiers" param directly. You can use the `Strings` type to pass a list of them then space them appropriately. That way you can process the shady IR annotations in `emit_decl`, and pass an empty list where we generate auxiliary global variables from "thin air"